src/axom/quest/interface/inout.cpp

```diff
@@ -401,13 +401,21 @@ int inout_init(mint::Mesh*& mesh, MPI_Comm comm)
 int inout_finalize()
 {
   const int dim = inout_get_dimension();
+  SLIC_ASSERT(dim == 2 || dim == 3);
 
-  // Finalize the 2D and 3D structures and reset the params
-  int rc2 = s_inoutHelper2D.finalize();
-  int rc3 = s_inoutHelper3D.finalize();
```

Comment:
The problem might have been calling finalize in 2D \*and\* 3D when it was only initialized in 2D \*or\* 3D.